### PR TITLE
feat: Form `rule` support `warningOnly` that not block form submit

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://preview-30829-ant-design.surge.sh/components/form-cn/#components-form-demo-warning-only

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by BigCkam34578** [comment]
[]()

File: components/_util/motion.tsx [hunk 1]
```diff
@@ -3,7 +3,17 @@ import { MotionEvent } from 'rc-motion/lib/interface';
 
 // ================== Collapse Motion ==================
 const getCollapsedHeight: MotionEventHandler = () => ({ height: 0, opacity: 0 });
-const getRealHeight: MotionEventHandler = node => ({ height: node.scrollHeight, opacity: 1 });
+const getRealHeight: MotionEventHandler = node => {
+  const { overflow } = node.style;
```
Comment (yoyo837) [inline components/_util/motion.tsx:7]: 这会不会有overflow-x, overflow-y 的属性？

image

Comment (zombieJ) [reply components/_util/motion.tsx:7]: 这个 `overflow` 只做覆盖还原用，不在乎它的值的~

**Comment by zombieJ** [comment]
TS 校验那个需要 antd 发包才能对上，先不用管

**Review by hengkx — APPROVED** [review]

**Review by xrkffgg — APPROVED** [review]

File: components/_util/motion.tsx (hunk 1 above)
Comment (yoyo837) [reply components/_util/motion.tsx:7]: 会给其他值怼没了（忽略我默认值是否合理）

image

Comment (yoyo837) [reply components/_util/motion.tsx:7]: 或者看这个

image

Comment (yoyo837) [reply components/_util/motion.tsx:7]: 这是否也是预期的？

Comment (zombieJ) [reply components/_util/motion.tsx:7]: 这个代码应该没用了，我去掉它~

**Comment by afc163** [comment]
ci 挂了

**Comment by zombieJ** [comment]
> ci 挂了

`TS 校验那个需要 antd 发包才能对上，先不用管`

**Comment by afc163** [comment]
测试覆盖率掉了 3 行。

**Pull request merged**